### PR TITLE
Decouple Tracking Code Generator from Overlay plugin

### LINKED ISSUE #21891: [Bug] Tracking Code requires Overlay plugin to be enabled
### What happened?

With Overlay plugin deactivated, and you go to 

Measureables -> Tracking Code

You get a warning:

> The plugin Overlay is not enabled. You can activate the plugin on Settings > Plugins page in Matomo.

It also makes the area to get the Tracking Code for JavaScript non working (empty).

### What should happen?

You should get the Tracking Code, and no warning.

### How can this be reproduced?

Deactivate Overlay plugin if it is enabled.

Then you go to:
Measureables -> Tracking Code

You get the warning and nothing for the Tracking Code.

### Matomo version

5.0.2

### PHP version

_No response_

### Server operating system

_No response_

### What browsers are you seeing the problem on?

_No response_

### Computer operating system

_No response_

### Relevant log output

```shell
Uncaught exception in API: /var/www/html/core/Plugin/Manager.php(303): The plugin Overlay is not enabled. You can activate the plugin on Settings > Plugins page in Matomo. [Query: ?date=yesterday&module=API&format=json&method=Overlay.getExcludedQueryParameters&idSite=1&filter_limit=-1&segment=&period=day, CLI mode: 0]
```

### Validations

- [X] Read our [Contributing Guidelines](https://github.com/matomo-org/matomo/blob/5.x-dev/CONTRIBUTING.md).
- [X] Follow our [Security Policy](https://github.com/matomo-org/matomo/blob/5.x-dev/SECURITY.md).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating duplicates.
- [X] The provided steps to reproduce is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the Bug.

### PR DESCRIPTION
**Pull request by sgiehl**
### Description:

The Tracking Code generator currently relies on using the API method `Overlay.getExcludedQueryParameters`. If that plugin is disabled, the API method is unavailable causing the generator to fail.

This PR deprecates the API method `Overlay.getExcludedQueryParameters` and introduces `SitesManager.getExcludedQueryParameters` as replacement.

fixes #21891

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,12 @@ This is the Developer Changelog for Matomo platform developers. All changes in o
 
 The Product Changelog at **[matomo.org/changelog](https://matomo.org/changelog)** lets you see more details about any Matomo release, such as the list of new guides and FAQs, security fixes, and links to all closed issues. 
 
+## Matomo 5.1.0
+
+### Deprecations
+
+The API method `Overlay.getExcludedQueryParameters` has been deprecated and will be remove with Matomo 6. Use the new method `SitesManager.getExcludedQueryParameters` instead.
```
Comment (michalkleiner) [inline CHANGELOG.md]:
```suggestion
The API method `Overlay.getExcludedQueryParameters` has been deprecated and will be removed in Matomo 6. Use the new method `SitesManager.getExcludedQueryParameters` instead.
```

File: plugins/Overlay/API.php [hunk 2]
```diff
@@ -40,19 +41,13 @@ public function getTranslations($idSite)
     /**
      * Get excluded query parameters for a site.
      * This information is used for client side url normalization.
+     *
+     * @deprecated use SitesManager.getExcludedQueryParameters instead
+     * @todo Remove with Matomo 6
```
Comment (michalkleiner) [inline plugins/Overlay/API.php]:
```suggestion
     * @todo Remove in Matomo 6
```

**Comment by michalkleiner** [comment]
Not a full review, just passed by and suggested a wording tweak.

File: plugins/Overlay/API.php [hunk 3]
```diff
@@ -40,19 +41,13 @@ public function getTranslations($idSite)
...
-        $site = $sitesManager->getSiteFromId($idSite);
-
-        try {
-            return SitesManager::getTrackerExcludedQueryParameters($site);
-        } catch (Exception $e) {
-            // an exception is thrown when the user has no view access.
-            // do not throw the exception to the outside.
-            return array();
-        }
+        return Request::processRequest('SitesManager.getExcludedQueryParameters');
```
Comment (michalkleiner) [inline plugins/Overlay/API.php:50]: Do we need to pass the parameter to maintain compatibility until the method is removed if someone somewhere still uses this (now deprecated) method?
```suggestion
        return Request::processRequest('SitesManager.getExcludedQueryParameters', ['idSite => $idSite']);
```
or
```suggestion
        return Request::processRequest('SitesManager.getExcludedQueryParameters', $idSite);
```

Comment (sgiehl) [reply plugins/Overlay/API.php:50]: No. `Request::processRequest` by default uses all request parameters from the original request. So as we want to perform another API action with the exact same parameter set, there is no need to overwrite any parameter.

Comment (michalkleiner) [reply plugins/Overlay/API.php:50]: Thanks for the explanation :)

**Review by michalkleiner — APPROVED** [review]

**Pull request merged**